core/tests/protractor/learnerFlow.js

```diff
@@ -0,0 +1,411 @@
+      // Play exploration 'Dummy Exploration'
+      libraryPage.get();
+      libraryPage.findExploration('Dummy Exploration');
+      libraryPage.playExploration('Dummy Exploration');
+      explorationPlayerPage.expectExplorationNameToBe('Dummy Exploration');
+
+      // Play exploration 'About Oppia'.
+      libraryPage.get();
+      libraryPage.findExploration('About Oppia');
+      libraryPage.playExploration('About Oppia');
+      explorationPlayerPage.expectExplorationNameToBe('About Oppia');
+      explorationPlayerPage.submitAnswer('Continue', null);
+      explorationPlayerPage.expectExplorationToNotBeOver();
+
+      // User clicks on Oppia logo to leave exploration.
+      oppiaLogo.click();
+      general.acceptAlert();
+      // Wait for /learner_dashboard to load.
+      waitFor.pageToFullyLoad();
+
+      // Learner Dashboard should display 'About Oppia'
+      // exploration as incomplete.
+      checkIncompleteExplorationSection('About Oppia');
+      // Learner Dashboard should display 'Dummy Exploration' as complete.
+      checkCompleteExplorationSection('Dummy Exploration');
+
+      // Now play exploration 'About Oppia' completely.
+      libraryPage.get();
+      libraryPage.findExploration('About Oppia');
+      libraryPage.playExploration('About Oppia');
+      waitFor.pageToFullyLoad();
+      explorationPlayerPage.expectExplorationNameToBe('About Oppia');
+      explorationPlayerPage.submitAnswer('Continue', null);
+      explorationPlayerPage.submitAnswer(
+        'MultipleChoiceInput', 'Those were all the questions I had!');
+
+      // Both should be added to the completed section.
+      learnerDashboardPage.get();
+      checkCompleteExplorationSection('About Oppia');
+      checkCompleteExplorationSection('Dummy Exploration');
+      users.logout();
+
+      // Login as Admin and delete exploration 'About Oppia'.
+      users.createAndLoginAdminUser('inspector@learnerDashboard.com',
+        'inspector');
+      libraryPage.get();
+      libraryPage.findExploration('About Oppia');
+      libraryPage.playExploration('About Oppia');
+      // Wait for player page to completely load
+      waitFor.pageToFullyLoad();
+      general.getExplorationIdFromPlayer().then(function(explorationId) {
+        general.openEditor(explorationId);
+      });
+      explorationEditorPage.navigateToSettingsTab();
+      explorationEditorSettingsTab.deleteExploration();
+      users.logout();
+
+      // Verify exploration 'About Oppia' is deleted from learner dashboard.
+      users.login('learner@learnerDashboard.com');
+      learnerDashboardPage.get();
+      learnerDashboardPage.navigateToCompletedSection();
+      learnerDashboardPage.expectTitleOfExplorationSummaryTileToMatch(
+        'Dummy Exploration');
+      learnerDashboardPage.expectTitleOfExplorationSummaryTileToBeHidden(
+        'About Oppia');
+    }
+  });
+
+  it('displays incomplete and completed collections', function() {
+    if (browser.isMobile) {
```

Comment:
Ditto. Only the exp ID should branch, nothing else.

Reply:
Done, thanks!